### ISSUE TITLE
Using a custom text vectorizer?

### ISSUE THREAD

**Issue by neil-119** [dotnet/machinelearning-samples#702]
I come from a Keras background, but deal mainly in C#, so I'm quite interested in this project. How can we use a custom vectorizer like word2vec for our text classification problems? We're getting accuracy around 60% with AutoML, whereas a simple two layer dense/fully-connected NN using the output from an e.g. [pre-trained word2vec model](http://vectors.nlpl.eu/explore/embeddings/en/models/) can boost accuracy from 60% to 100% in our dataset. While throwing more data would certainly help, it is not feasible to do so right now, so I'd like to use a pre-trained vectorizer instead.

**Comment by justinormont** [comment]
You can use a custom word embedding if you created a domain specific one, or use the built-in fastText or GloVe models. I don't see a sample of the Word Embedding transform.  (@CESARDELATORRE -- would could use a word embedding sample).

If you're using the AutoML API you can place the Word Embedding transform in the [preFeaturizer](https://github.com/dotnet/machinelearning-samples/blob/2a86b9aac3af9c4eeab7655e7869f02132c0de57/samples/csharp/getting-started/AdvancedExperiment_AutoML/AdvancedTaxiFarePrediction/Program.cs#L99-L105). Then AutoML will optimize the learners with this added featurization.

Your preFeaturizer will be approximately:
```C#
var featurizeTextOptions = new TextFeaturizingEstimator.Options() { OutputTokensColumnName = "Tokens" };
            
var pipeline = ML.Transforms.Text.FeaturizeText("FeaturesNGrams", featurizeTextOptions, "TextColumn")
    .Append(ML.Transforms.Text.ApplyWordEmbedding("FeaturesWordEmbeddings", "Tokens", modelKind: WordEmbeddingEstimator.PretrainedModelKind.FastTextWikipedia300D));
```

If you're using the AutoML CLI or Model Builder, you can add the Word Embedding transform to the created code. Though since it's added afterwards, the model is not optimized towards the word embedding features.

Ideally, AutoML would be iterating over the choice of using a word embedding or not. Though it does not currently do this.

**Comment by luisquintanilla** [comment]
Closing this issue since a response was provided and there has been no additional activity
